drivers/spi/spi_ll_stm32.c

```diff
@@ -338,20 +338,57 @@ static int spi_stm32_get_err(SPI_TypeDef *spi)
 	return 0;
 }
 
-/* Shift a SPI frame as master. */
-static void spi_stm32_shift_m(SPI_TypeDef *spi, struct spi_stm32_data *data)
+static int spi_stm32_can_use_fifo(SPI_TypeDef *spi,
+								  struct spi_stm32_data *data)
+{
+	ARG_UNUSED(spi);
+
+#if SPI_STM32_FIFO_ENABLED
```

Comment:
Why not using `#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7_spi)` directly ?
H7 compatible SPI IPs are variants of SPI-FIFO IPs

Reply:
Normally, the actual check should be only if the SPI device in question has a FIFO (indicated in devicetree). However, I add the H7 check as well as that's the only platform I can test. I could leave the H7 check only, but I wanted to remark that the actual criteria to enable this (or not) is whether or not a FIFO is available, so I left both checks so another user can (hopefully) understand the intention.

Reply:
This is misleading as using `SPI_STM32_FIFO_ENABLED`, it's hard to guess that code impacts only H7 compatible devices.
Question is: Is that code compatible with all `st,stm32-spi-fifo` compatible devices ? or is that specific to `st,stm32h7-spi` ones ?

Reply:
> Question is: Is that code compatible with all st,stm32-spi-fifo compatible devices ? or is that specific to st,stm32h7-spi ones ?

I can't answer this question because I can only test this change for H7. It _should_ be compatible with `st,stm32-spi-fifo`, but I can't assure it as I can't test it. I want to restrict the change to H7 to avoid introducing bugs or breaking applications already using Zephyr.

However, I can add a comment to explain the purpose of checking H7.

Reply:
> However, I can add a comment to explain the purpose of checking H7.

Please do the opposite:
Use a check on `st,stm32h7-spi`. Add a comment saying it has to be verified on `st,stm32-spi-fifo`.
It will be much easier to understand quickly the situation.